Add a command to enable and disable the autoplaylist

Zenrac opened the pull request:
I made a command to enable/disable the autoplaylist without restarting the bot.

I think it can be very useful ;)

Zenrac committed:
A command to enable/disable the autoplaylist

A made a command to enable/disable the autoplaylist without restarting the bot.

Zenrac committed:
Update bot.py

Zenrac committed:
Update bot.py

jayktaylor commented on musicbot/bot.py:
```diff
@@ -1950,6 +1950,27 @@ def check(m):
+            self.config.auto_playlist = True
+            if not player.playlist.entries and not player.current_entry and self.config.auto_playlist: #if nothing is queued, start a song
+                await player.playlist.add_entry(song_url, channel=None, author=None)
+            await self.safe_send_message(channel, "The autoplaylist is now enabled!")
```
This should return a `Response` object instead, along with the same line in the section below.

jayktaylor commented on musicbot/bot.py:
```diff
@@ -1950,6 +1950,27 @@ def check(m):
+            self.config.auto_playlist = False
+            await self.safe_send_message(channel, "The autoplaylist is now disabled!")
+            return							
+	
```
RE: The whole thing, it may just make sense to invert the `self.config.auto_playlist` boolean and then do nothing else. That'll mean that after the current queue has finished playing, `on_player_finished_playing()` should be triggered and detect that the autoplaylist is enabled.

This is probably the most simple implementation:

```py
self.config.auto_playlist = not self.config.auto_playlist
return Response("The autoplaylist is now " + ['disabled', 'enabled'][self.config.auto_playlist])

Zenrac committed:
simplified main code

Zenrac committed:
Fixed

jayktaylor commented on musicbot/bot.py:
```diff
@@ -1960,7 +1960,7 @@ def check(m):
         self.config.auto_playlist = not self.config.auto_playlist
         await self.safe_send_message(channel, "The autoplaylist is now " + ['disabled', 'enabled'][self.config.auto_playlist])
```
Send this message to Discord *last*, while returning a Response object as previously mentioned.

jayktaylor commented on musicbot/bot.py:
```diff
@@ -1960,7 +1960,7 @@ def check(m):
         if not player.playlist.entries and not player.current_entry and self.config.auto_playlist: #if nothing is queued, start a song
-            song_url = random.choice(player.autoplaylist) 
+            song_url = random.choice(self.autoplaylist) 
             await player.playlist.add_entry(song_url, channel=None, author=None)							
```
Rather than repeating what is already in the `on_player_finished_playing` function, this whole section (line 1962-1964) could be simplified even further by:

```py
if self.config.auto_playlist:
    await self.on_player_finished_playing(player)
```

Zenrac committed:
Update bot.py

Zenrac committed:
Optional : Now it checks if the autoplaylist is empty

if you didn't remove the "auto disable" of the autoplaylist when it is empty it can be useful, otherwise don't mind this changes.

Zenrac committed:
Update bot.py

Zenrac committed:
Little update (I was tired?)
